Procfile

```diff
@@ -1,2 +1,3 @@
-web: vendor/bin/heroku-php-nginx -C nginx.conf public/
-release: if [ -n "$DB_HOST" ]; then php artisan migrate --force; fi
+web: composer warmup && $(composer config bin-dir)/heroku-php-nginx -C nginx.conf public/
+scheduler: php -d memory_limit=512M artisan queue:work --stop-when-empty
+release: php artisan migrate --force && php artisan cache:clear
```

Comment:
you want to support people pushing an app that requires a database, without a database... The app.json forces a database creation meaning the check is not needed. Do you wish for this not to deploy a database?

Reply:
ah I didn't realize that was the case, so you're saying Heroku will always create a database when this is pushed now that app.json defines that? I'm just trying to make sure that this still works if someone manually does a git clone and git push instead of using the deploy button.

Reply:
In the case of a git push, there would be no database as they do not read app.json or automatically setup services.

One small issue is that without a database I think all these services deploy, but die. If that is the goal, might I suggest sqlite as a default, and sharding the schema, or putting some code in to skip migrations if no database. I could throw in an ` || echo "unable to complete release"`

Reply:
Okay that was my fear, and why I had the "if" in there before. The case to support here is someone does a `git clone` from github, pushes to a heroku branch before setting up a database add-on, and then heroku runs the `release` script and it has to not crash. As long as this won't halt the deploy it's fine, so `|| echo` might be the easiest option. I'd suggest a message that better explains what's happening though, such as "skipping migrations because no database is configured"

Reply:
Sorry, but it's not skipping for that reason. It's a failed release. You're in a funny situation if we use `||` as it won't differentiate on failure reason.

* you may not have access to ENV
* you may not have a database
* you may in-fact have other issues

It will not affect logging, so you will have a full history, but it won't guard production if you push trash

Reply:
That is why I setup a database in the app.json, but the desire to have non heroku-button manual deploys does raise issues.

Reply:
It's not so much a desire as the fact that plenty of people will do it regardless of whether we want them to, and it shouldn't block them from doing that.